### PR TITLE
Add Mac Catalyst RIDs

### LINKED ISSUE #47518: Add new RID for Catalyst
We need to introduce new Catalyst RIDs to be able to build Catalyst runtime packs. I propose we add the following new RIDs to support them inheriting from iOS RID.

`maccatalyst-arm64`
`maccatalyst-arm64-13`
`maccatalyst-arm64-14`
`maccatalyst-x64`
`maccatalyst-x64-13`
`maccatalyst-x64-14`

I went with a simple catalyst instead of macOSCatalyst as it's very long and it runs on iPadOS as well.

@steveisok @terrajobst

### PR DESCRIPTION
**Pull request by steveisok**
First step in being able to build different mac catalyst runtimes

Resolves https://github.com/dotnet/runtime/issues/47518

### CODE REVIEW

File: src/libraries/Microsoft.NETCore.Platforms/pkg/runtime.json [hunk 1]
```diff
@@ -560,6 +560,57 @@
         "browser"
       ]
     },
+    "catalyst": {
+      "#import": [
+        "linux"
```
Comment (lambdageek) [inline src/libraries/Microsoft.NETCore.Platforms/pkg/runtime.json:565]: (I don't know how RIDs work)

Why is it based on "linux"?  "osx" and "ios" are both "unix".  Shouldn't catalyst be like them?

Comment (steveisok) [reply src/libraries/Microsoft.NETCore.Platforms/pkg/runtime.json:565]: Yes, that's what I get for speed reading Marek's issue ;-)

**Comment by safern** [comment]
cc: @ericstj

**Comment by akoeplinger** [comment]
Note: there's still some discussion ongoing in https://github.com/dotnet/runtime/issues/47518 about whether we should use `maccatalyst` as the RID instead.

**Comment by ericstj** [comment]
cc @directhex @Anipik we also have a simplified RID graph that we use for building libraries in dotnet/runtime here: https://github.com/dotnet/runtime/blob/6fe449fefcd6c3a850598bee62ca22c79961fd6e/src/libraries/OSGroups.json#L17.  You may want to update this as well if you intend to add `maccatalyst` build of dotnet/runtime.

**Review by ericstj — APPROVED** [review]
Thanks for the discussion, this LGTM

**Pull request merged**